src/librustdoc/html/render/print_item.rs

```diff
@@ -263,11 +263,30 @@ fn item_module(w: &mut Buffer, cx: &Context<'_>, item: &clean::Item, items: &[cl
             }
 
             clean::ImportItem(ref import) => {
+                let mut stab = None;
+                let mut stab_tags = None;
+
+                if let Some(did) = import.source.did {
+                    // Just need an item with the correct def_id
+                    let mut import_item = myitem.clone();
+                    import_item.def_id = did;
+                    stab = import_item.stability_class(cx.tcx());
+                    stab_tags = Some(extra_info_tags(&import_item, item, cx.tcx()));
+                }
```

Comment:
nit: these don't need to be mutable.
```suggestion
                let (stab, stab_tags) = if let Some(def_id) = import.source.did {
                    // Just need an item with the correct def_id
                    let import_item = Item { def_id, ..myitem.clone() };
                    let stab = import_item.stability_class(cx.tcx());
                    let stab_tags = Some(extra_info_tags(&import_item, item, cx.tcx()));
                    (stab, stab_tags)
                } else {
                    (None, None)
                };
```